### PR TITLE
feat: Track upload progress

### PR DESCRIPTION
**Pull request by sirambd**

### CODE REVIEW

**Review by tevincent — APPROVED** [review]

**Review by tevincent — APPROVED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/workers/UploadWorker.kt [hunk 1]
```diff
@@ -136,6 +166,8 @@ class UploadWorker @AssistedInject constructor(
         private const val MAX_CHUNK_COUNT = (TOTAL_FILE_SIZE / EXPECTED_CHUNK_SIZE).toInt()
 
         private const val UPLOADED_BYTES_TAG = "uploaded_bytes_tag"
-        private const val TOTAL_SIZE_TAG = "total_size_tag"
+        private const val TRANSFER_UUID_TAG = "transfer_result_tag"
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/swisstransfer/workers/UploadWorker.kt:169]: Is the actual string key `"transfer_result_tag"` what you actually wanted?

**Review by LunarX — CHANGES_REQUESTED** [review]

**Comment by LunarX** [comment]
I once managed to reach this error when clicking on "Send" after building commit `c897300`. Does it ring a bell?

```
13:25:26.041 System.err            W  java.lang.IllegalStateException: The file /data/user/0/com.infomaniak.swisstransfer/files/imported_files/anime-iphone-background-psdmm565oizldbbg.jpg doesn't exists
13:25:26.042 System.err            W  	at com.infomaniak.swisstransfer.workers.UploadFileTask$start$2.invokeSuspend(UploadFileTask.kt:154)
13:25:26.042 System.err            W  	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
13:25:26.042 System.err            W  	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:101)
13:25:26.042 System.err            W  	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:589)
13:25:26.042 System.err            W  	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:832)
13:25:26.042 System.err            W  	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:720)
13:25:26.042 System.err            W  	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:707)
```

File: app/src/main/java/com/infomaniak/swisstransfer/workers/UploadFileTask.kt [hunk 2]
```diff
@@ -125,6 +126,7 @@ class UploadFileTask(
             data = data,
             onUpload = { bytesSentTotal, _ ->
                 mutex.withLock {
+                    ensureActive()
```
[the remark this answers is not in the document]
Comment (sirambd) [reply app/src/main/java/com/infomaniak/swisstransfer/workers/UploadFileTask.kt:129]: To close the coroutine writing to the server

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadProgressScreen.kt [hunk 3]
```diff
@@ -37,31 +34,41 @@ import com.infomaniak.swisstransfer.R
...
-
     var showBottomSheet by rememberSaveable { mutableStateOf(false) }
 
     BackHandler(!showBottomSheet) {
         showBottomSheet = true
     }
 
+    LaunchedEffect(Unit) {
+        uploadProgressViewModel.trackUploadProgress()
+    }
```
[the remark this answers is not in the document]
Comment (sirambd) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadProgressScreen.kt:65]: No, it's the good practice

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt [hunk 4]
```diff
@@ -58,12 +59,15 @@ private const val TOTAL_FILE_SIZE: Long = 50_000_000_000L
...
-    navigateToUploadProgress: () -> Unit,
+    navigateToUploadProgress: (transferType: TransferType, totalSize: Long) -> Unit,
 ) {
     val files by newTransferViewModel.importedFilesDebounced.collectAsStateWithLifecycle()
     val filesToImportCount by newTransferViewModel.filesToImportCount.collectAsStateWithLifecycle()
     val currentSessionFilesCount by newTransferViewModel.currentSessionFilesCount.collectAsStateWithLifecycle()
     val selectedTransferType by newTransferViewModel.selectedTransferType.collectAsStateWithLifecycle()
+    val sendActionResult by newTransferViewModel.sendActionResult.collectAsStateWithLifecycle()
+
+    HandleSendActionResult({ sendActionResult }, selectedTransferType, navigateToUploadProgress)
```
[the remark this answers is not in the document]
Comment (sirambd) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt:70]: Not enough to avoid recomposition

**Review by LunarX — APPROVED** [review]

**Pull request merged**